src/plugins/discover/public/components/data_types/logs/service_name_cell.tsx

```diff
@@ -7,39 +7,49 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { EuiFlexGroup, EuiFlexItem, EuiToolTip } from '@elastic/eui';
+import React from 'react';
+import { EuiToolTip } from '@elastic/eui';
 import type { AgentName } from '@kbn/elastic-agent-utils';
 import { dynamic } from '@kbn/shared-ux-utility';
 import type { DataGridCellValueElementProps } from '@kbn/unified-data-table';
-import React from 'react';
+import { css } from '@emotion/react';
 import { getFieldValue } from '@kbn/discover-utils';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { CellRenderersExtensionParams } from '../../../context_awareness';
 import { AGENT_NAME_FIELD } from '../../../../common/data_types/logs/constants';
+import { ServiceNameBadgeWithActions } from './service_name_badge_with_actions';
 
 const dataTestSubj = 'serviceNameCell';
 const AgentIcon = dynamic(() => import('@kbn/custom-icons/src/components/agent_icon'));
 
 export const getServiceNameCell =
-  (serviceNameField: string) => (props: DataGridCellValueElementProps) => {
+  (serviceNameField: string, { actions }: CellRenderersExtensionParams) =>
+  (props: DataGridCellValueElementProps) => {
     const serviceNameValue = getFieldValue(props.row, serviceNameField) as string;
     const agentName = getFieldValue(props.row, AGENT_NAME_FIELD) as AgentName;
 
     if (!serviceNameValue) {
       return <span data-test-subj={`${dataTestSubj}-empty`}>-</span>;
     }
 
+    const getIcon = () => (
+      <EuiToolTip position="left" content={agentName} repositionOnScroll={true}>
+        <AgentIcon
+          agentName={agentName}
+          size="m"
+          css={css`
+            margin-right: ${euiThemeVars.euiSizeXS};
```

Comment:
I see this but also see the icon bring right up against the text. Is this what this line is aiming to avoid or do we need to add margin here too? (see image)
![CleanShot 2024-10-21 at 07 30 35@2x](https://github.com/user-attachments/assets/206f7a91-fe4a-40c9-ac11-9ea71b9d5579)

Reply:
Nit: since these styles are the same across instances, declaring them as a constant outside of the component would be more performant. I usually wouldn't recommend such a micro optimization, but it's worth considering since performance can be very sensitive within grid cells where we render so many at once.